### PR TITLE
Secure connexion between TinyBird and webhookResponseGraph

### PR DESCRIPTION
**Pull request by anamarn**
TLDR:
Secure connexion between tinybird and twenty using jwt when accessing datasource from tinybird.

Solves:
https://github.com/twentyhq/private-issues/issues/73

In order to test:

1. Set ANALYTICS_ENABLED to true
2. Set TINYBIRD_JWT_TOKEN to the ADMIN token from the workspace twenty_analytics_playground
3. Set TINYBIRD_JWT_TOKEN to the datasource or your admin token from the workspace twenty_analytics_playground
4. Create a Webhook in twenty and set wich events it needs to track
5. Run twenty-worker in order to make the webhooks work.
6. Do your tasks in order to populate the data
7. Enter to settings> webhook>your webhook and the statistics section should be displayed.

### CODE REVIEW

File: packages/twenty-front/src/modules/settings/developers/webhook/hooks/useAnalyticsTinybirdJwt.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { useRecoilValue } from 'recoil';
+
+import { currentUserState } from '@/auth/states/currentUserState';
+
+export const useAnalyticsTinybirdJwt = (): string | null | undefined => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/developers/webhook/hooks/useAnalyticsTinybirdJwt.ts:5]: can it be null

File: packages/twenty-front/src/modules/settings/developers/webhook/hooks/useGraphData.tsx [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
   const { enqueueSnackBar } = useSnackBar();
+  const supportTinybirdJwt = useAnalyticsTinybirdJwt();
   const fetchGraphData = async (
     windowLengthGraphOption: '7D' | '1D' | '12H' | '4H',
   ) => {
     try {
       return await fetchGraphDataOrThrow({
         webhookId,
         windowLength: windowLengthGraphOption,
+        tinybirdJwt: supportTinybirdJwt ?? undefined,
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/developers/webhook/hooks/useGraphData.tsx:16]: ?? undefined is not useful, we should throw before if the jwt is undefined

File: packages/twenty-server/src/engine/core-modules/analytics/analytics.module.ts [hunk 3]
```diff
@@ -1,17 +1,34 @@
+/* eslint-disable no-restricted-imports */
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/analytics/analytics.module.ts:1]: why?

File: packages/twenty-server/src/engine/core-modules/analytics/analytics.service.ts [hunk 4]
```diff
@@ -58,7 +60,7 @@ export class AnalyticsService {
     const config: AxiosRequestConfig = {
       headers: {
         Authorization:
-          'Bearer ' + this.environmentService.get('TINYBIRD_TOKEN'),
+          'Bearer ' + this.environmentService.get('TINYBIRD_DATASOURCE_TOKEN'),
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/analytics/analytics.service.ts:63]: TINYBIRD_INGEST_TOKEN
(vs TINYBIRD_GENERATE_JWT_TOKEN)

File: packages/twenty-server/src/engine/core-modules/analytics/analytics.service.ts [hunk 5]
```diff
@@ -86,4 +88,22 @@ export class AnalyticsService {
 
     return { success: true };
   }
+
+  async generateJWT(workspaceId: string | null | undefined) {
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/analytics/analytics.service.ts:92]: generateWorkspaceJwt

File: packages/twenty-server/src/engine/core-modules/environment/environment-variables.ts [hunk 6]
```diff
@@ -91,11 +91,23 @@ export class EnvironmentVariables {
   @CastToBoolean()
   @IsOptional()
   @IsBoolean()
-  ANALYTICS_ENABLED = false;
+  ANALYTICS_ENABLED = true;
 
   @IsString()
   @ValidateIf((env) => env.ANALYTICS_ENABLED)
-  TINYBIRD_TOKEN: string;
+  TINYBIRD_DATASOURCE_TOKEN: string;
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/environment/environment-variables.ts:98]: add these to doc

File: packages/twenty-front/src/modules/settings/developers/webhook/hooks/useAnalyticsTinybirdJwt.ts (hunk 1 above)
Comment (anamarn) [reply packages/twenty-front/src/modules/settings/developers/webhook/hooks/useAnalyticsTinybirdJwt.ts:5]: yes it can be null when ANALYTICS is not enabled in the server

**Review by charlesBochet — APPROVED** [review]
